storage/connectors/redis/src/main/java/feast/storage/connectors/redis/writer/RedisCustomIO.java

```diff
@@ -307,6 +375,49 @@ boolean rowShouldBeWritten(FeatureRow newRow, byte[] currentValue) {
             .map(e -> ImmutablePair.of(e.getKey(), Iterators.getLast(e.getValue().iterator())))
             .collect(Collectors.toMap(ImmutablePair::getLeft, ImmutablePair::getRight));
       }
+
+      // Helper function for calculating the TTL for the redis row.
+      // Return true if the row should be added and false otherwise.
+      private static boolean setRedisTtl(
+          FeatureSetSpec spec,
+          FeatureRow newRow,
+          RedisKey key,
+          Map<RedisKey, Long> keysWithTtl,
+          List<FeatureRow> skippedFeatureRows,
+          int maxRedisTtlJitterSeconds) {
+        final com.google.protobuf.Duration maxAge = spec.getMaxAge();
+
+        // If maxAge isn't set don't apply TTL
+        if (!maxAge.equals(com.google.protobuf.Duration.getDefaultInstance())) {
+          long ttlSeconds = maxAge.getSeconds();
+
+          // Adjust initial TTL based on event timestamp of
+          // FeatureRow
+          ttlSeconds -=
+              currentTime.get().toInstant().getEpochSecond()
+                  - newRow.getEventTimestamp().getSeconds();
+
+          if (ttlSeconds > 0) {
+            // The consideration to write data to Redis or not
+            // should NOT take jitter into account
+            if (maxRedisTtlJitterSeconds > 0) {
+              ttlSeconds += rand.nextInt(maxRedisTtlJitterSeconds);
+            }
+            keysWithTtl.put(key, ttlSeconds);
+          } else {
+            skippedFeatureRows.add(newRow);
+            return false;
+          }
+        } else {
+          // Log an error instead of throwing an exception since
+          // setting a TTL is an
+          // optimization
+          // for storage space usage.
+          log.error("Unable to find FeatureSet to set Redis TTL  featureSet={}", spec.getName());
```

Comment:
Does this mean we will always log errors for users that have feature sets with default max age of 0?

Reply:
No, it shouldn't. The intention is that we'll log if no max age is set at all.

Reply:
Not setting a max age is actually very common. What if we only log a warning here, and we only log if neither a max age has been configured on the feature set or as part of the redis config (maximum max age).

I think as administrators we must be able to set the expiration, even if users dont set the max age.